be/src/runtime/memory/thread_mem_tracker_mgr.cpp

```diff
@@ -29,25 +29,25 @@ void ThreadMemTrackerMgr::attach_limiter_tracker(
         const std::shared_ptr<MemTrackerLimiter>& mem_tracker) {
     DCHECK(mem_tracker);
     flush_untracked_mem<false>();
-    _task_id_stack.push_back(task_id);
-    _fragment_instance_id_stack.push_back(fragment_instance_id);
-    _limiter_tracker_stack.push_back(mem_tracker);
+    _task_id = task_id;
+    _fragment_instance_id = fragment_instance_id;
```

Comment:
instance id is useless， we need cancel query by query id.

Reply:
done